app/src/androidTest/java/com/group80/uoftinder/UserRealtimeDbFacadeUnitTest.java

```diff
@@ -0,0 +1,140 @@
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+/**
+ * Instrumented Unit tests for UserRealtimeDbFacadeUnitTest, run on Android devices
+ */
+@RunWith(AndroidJUnit4.class)
+public class UserRealtimeDbFacadeUnitTest {
+    /**
+     * Test if the upload task can be done without error
+     * <p>
+     * FIXME: the test succeeded even though the Users were not uploaded. However the upload task
+     *  can be done outside the texting environment.
+     */
+    @Test
+    public void uploadUser_isCorrect() {
+        FirebaseDatabase.getInstance().getReference().child("users").child("test").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot snapshot) {
+                assertNotSame(snapshot.getChildrenCount(), 0);
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+                Assert.fail(error.getMessage());
+            }
+        });
+
+        User lisinan2 = new User("lisinan2");
+        lisinan2.setName("Lance Li");
+        lisinan2.setUserType("test");
+        lisinan2.setViewed(new ArrayList<>(Arrays.asList("a", "b", "c")));
+
+        User vgvg = new User("vgvg");
+        vgvg.setName("Vedant Goel");
+        vgvg.setUserType("test");
+        vgvg.setViewed(new ArrayList<String>());
+        vgvg.setMatches(new ArrayList<>(Arrays.asList("d", "e", "f")));
+
+        RealtimeDbWriteListener listener = new RealtimeDbWriteListener() {
+            @Override
+            public void onWriteSuccess(Void unused) {
+            }
+
+            @Override
+            public void onWriteFailure(Exception e) {
+                Assert.fail(e.getMessage());
+            }
+        };
+
+        UserRealtimeDbFacade.uploadUser(lisinan2, listener);
+        UserRealtimeDbFacade.uploadUser(vgvg, listener);
+
+    }
+
+    /**
+     * The realtime database must contain the following entries:
+     * - users
+     * <p>
+     * _ - test
+     * <p>
+     * _ _ - lisinan2
+     * <p>
+     * _ _ _ - firstName: "Lance"
+     * <p>
+     * _ _ _ - lastName:  "Li"
+     * <p>
+     * _ _ _ - uid:       "lisinan2"
+     * <p>
+     * _ _ _ - userType:  "test"
+     * <p>
+     * _ _ - vgvg
+     * <p>
+     * _ _ _ - firstName: "Vedant"
+     * <p>
+     * _ _ _ - lastName:  "Goel"
+     * <p>
+     * _ _ _ - uid:       "vgvg"
+     * <p>
+     * _ _ _ - userType:  "test"
+     */
+    @Test
+    public void getAllUsers_isCorrect() {
+        StringBuilder namesBuilder = new StringBuilder();
+        UserRealtimeDbFacade.getAllUsers("test", userList -> {
+            userList.forEach(user -> namesBuilder.append(user.getUid()).append("_"));
+            // this line is equivalent to the following code chunk -- please refer to Java lambda expressions for more details
```

Comment:
Glad we talked this one out offline - originally believed we needed to call `new RealtimeDbCallback` to get this to work but Java lambda expressions is a nice & quick substitute